warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java

```diff
@@ -389,14 +392,31 @@ protected IndexInfo intersect(long maxPossibilities, Iterable<IndexMatch> matchI
         }
         
         JexlNode newNode;
-        if (matches.size() > 1) {
+        if (matches.isEmpty()) {
+            JexlNodeSet nodes = new JexlNodeSet();
+            if (matchNode != null) {
+                nodes.add(matchNode);
+            }
+            if (!otherInfiniteNodes.isEmpty()) {
+                nodes.addAll(otherInfiniteNodes);
+            }
+            if (!delayedNodes.isEmpty()) {
+                nodes.addAll(delayedNodes);
+            }
+            newNode = TreeFlatteningRebuildingVisitor.flatten(JexlNodeFactory.createAndNode(nodes.getNodes()));
+            
+        } else if (matches.size() == 1) {
+            
+            newNode = TreeFlatteningRebuildingVisitor.flatten(matches.iterator().next().getNode());
```

Comment:
what happens to delayed node in this case?

Reply:
and in the case below?

Reply:
This code was removed and is now obsolete.